genesyscloud/util/files/util_files.go

```diff
@@ -179,49 +182,76 @@ func DownloadOrOpenFile(path string) (io.Reader, *os.File, error) {
 	var reader io.Reader
 	var file *os.File
 
-	_, err := os.Stat(path)
-	if err != nil {
-		_, err = url.ParseRequestURI(path)
-		if err == nil {
-			resp, err := http.Get(path)
```

Comment:
Had to refactor this function because a non-existent filepath would get to this line and barf with an error complaining about not having a protocol, which was a red herring. The real issue is that the local file didn't exist. I refactored this function so that we check if the path has a protocol and make the call first, then if not, check to ensure the filepath exists. Also added test cases.